app/src/main/java/com/example/plantsapp/presentation/ui/notification/TaskNotificationManager.kt

```diff
@@ -78,11 +85,29 @@ class TaskNotificationManager @Inject constructor(
         )
     }
 
+    private fun createCompletePendingIntent(task: Task, notificationId: Int): PendingIntent {
+        val completeIntent = Intent(context, NotificationBroadcastReceiver::class.java)
+            .apply {
+                this.notificationId = notificationId
+                taskId = task.id
+                taskFrequency = task.frequency
+                taskKey = TaskKeys.from(task).key
+            }
+        return PendingIntent.getBroadcast(
+            context,
+            completeRequestCode++,
```

Comment:
It is hard to fix such things in the future, when you create some functions and inside of them you change some out-of-scope value, how about make this function more clean and add this code into parameters of this function?